pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+        """
+        if namespace not in self.instances:
+            raise CIMError(CIM_ERR_INVALID_NAMESPACE,
+                           'Namespace %s not found for instances' % namespace)
+        return
+
+    def get_superclasses(self, cn, namespace):
+        """
+        Get list of superclasses from the class repository for the defined
+        classname in the namespace.
+
+        Returns in hiearchy order.
+        """
+        superclasses = []
+        if cn:
+            cnwork = cn
+            while cnwork:
+                cnsuper = self.classes[namespace][cnwork].superclass
+                if cnsuper:
+                    superclasses.append(cnsuper)
+                cnwork = cnsuper
+            superclasses.reverse()
+        return superclasses
+
+    def _get_subclass_names(self, classname, namespace, deep_inheritance=False):
+        """
+            Get class names that are subclasses of the
+            classname input parameter from the repository.
+
+            If DeepInheritance is False, get only classes in the
+            repository for the defined namespace for which this class is a
+            direct super class.
+
+            If deep_inheritance is True, get all direct and indirect
+            subclasses.
+
+        Returns:
+            list of strings defining the subclass names.
+
+        """
+        rtn_classnames = [cl.classname
+                          for cl in six.itervalues(self.classes[namespace])
+                          if cl.superclass == classname]
+
+        if deep_inheritance:
+            subclass_names = []
+            if rtn_classnames:
+                for cn in rtn_classnames:
+                    subclass_names.extend(
+                        self._get_subclass_names(
+                            cn, namespace,
+                            deep_inheritance=deep_inheritance))
+            rtn_classnames.extend(subclass_names)
+        return rtn_classnames
+
+    def _get_class(self, classname, namespace, **params):
+        """
+        Get class repository interface.  Gets the class defined by classname
+        from the repository, creates a copy, expands to include superclass
+        properties if not localonly and filters the class based on propertylist
+        and includeClassOrigin.
+
+        Parameters:
+
+          classname (:term:`string`):
+            Name of class to retrieve
+
+          namespace (:term:`string`):
+            Namespace from which to retrieve the class
+
+          params(Dictionary of keywords that determine filtering):
+            The keywords in this dictionary determine any filtering on the
+            class before being returned including LocalOnly, PropertyList,
+            IncludeQualifiers, and IncludeClassOrigin.
+
+        Returns:
+            class if found with superclass properties installed and
+            filtered per the keywords in params.
+        """
+        self._validate_classes_namespace(namespace)
```

Comment:
The `_fake_...()` methods calling this method already perform namespace validation. I think for this internal method, it would be ok to raise KeyError if thenamespace does not exist (i.e. simply access without checking). We also already do that for the classname, BTW.